helm-charts/common/ollama-service/templates/deployment.yaml

```diff
@@ -0,0 +1,130 @@
+          args:
+            - ollama serve & sleep 10 && ollama pull ${LLM_MODEL_ID}
+          volumeMounts:
+            # Running as non-root sets $HOME to /
+            - mountPath: /.ollama
+              name: model-volume
+
+      containers:
+        - name: {{ .Chart.Name }}
+          command: ["bash", "-c"]
+          args: ["ollama serve & sleep 10 && ollama run {{ .Values.LLM_MODEL_ID }} & wait"]
+          envFrom:
+            - configMapRef:
+                name: {{ include "ollama-service.fullname" . }}-config
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.service.targetPort }}
+              protocol: TCP
+          {{- if .Values.livenessProbe }}
+          livenessProbe:
+            {{- toYaml .Values.livenessProbe | nindent 12 }}
+          {{- end }}
+          {{- if .Values.readinessProbe }}
+          readinessProbe:
+            {{- toYaml .Values.readinessProbe | nindent 12 }}
+          {{- end }}
+          startupProbe:
+            httpGet:
+              # Ollama uses lazy loading for models, but this verifies the model has been downloaded
+              path: /v1/models/{{ .Values.LLM_MODEL_ID }}
+              port: http
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            # Running as non-root sets $HOME to /
+            - mountPath: /.ollama
+              name: model-volume
+            {{- with .Values.volumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+      volumes:
+        - name: model-volume
+          {{- if .Values.global.modelUsePVC }}
+          persistentVolumeClaim:
+            claimName: {{ .Values.global.modelUsePVC }}
+          {{- else if .Values.global.modelUseHostPath }}
+          hostPath:
+            path: {{ .Values.global.modelUseHostPath }}
+            type: Directory
+          {{- else }}
+          emptyDir: {}
+          {{- end }}
+        {{- with .Values.volumes }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- if .Values.evenly_distributed }}
+      topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: kubernetes.io/hostname
+          whenUnsatisfiable: ScheduleAnyway
+          labelSelector:
+            matchLabels:
+              {{- include "ollama-service.selectorLabels" . | nindent 14 }}
+      {{- end }}
```

Comment:
I think setting that would schedule this service to same node where the service using the downloaded model is, would make more sense, than just randomly spreading download service around the cluster.